lib/rubocop/cop/performance/intersection_check.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # In Ruby 3.1, `Array#intersect?` has been added.
+      #
+      # This cop identifies places where `(receiver & argument).any?`
+      # can be replaced by `receiver.intersect?(argument)`.
+      #
+      # The `receiver.intersect?(argument)` method is faster than
+      # `(receiver & argument).any?`.
+      #
+      # @example
+      #   # bad
+      #   (receiver & argument).present?
+      #   (receiver & argument).any?
+      #   (receiver & argument).blank?
+      #   (receiver & argument).empty?
+      #
+      #   # good
+      #   receiver.intersect?(argument)
+      #   !receiver.intersect?(argument)
+      class IntersectionCheck < Base
+        extend AutoCorrector
+        extend TargetRubyVersion
+
+        minimum_target_ruby_version 3.1
+
+        def_node_matcher :ampersand_intersection_check?, <<~PATTERN
+          (send
+            (begin 
+              (send $(...) :& $(...))
+            ) $/present?|any?|blank?|empty?/
+          )
+        PATTERN
+
+        MSG = 'Use `%<negated>s%<receiver>s.intersect?(%<argument>s)` ' \
+              'instead of `(%<receiver>s & %<argument>s).%<method_name>s`.'
+        STRAIGHT_METHODS = %i[present? any?].freeze
+        NEGATED_METHODS = %i[blank? empty?].freeze
```

Comment:
Rails dependent `present?` and `blank?` should only be targeted when `ActiveSupportExtensionsEnabled` is enabled:
https://docs.rubocop.org/rubocop/1.39/configuration.html#enable-checking-active-support-extensions